examples/soft-kmeans/README.md

```diff
@@ -0,0 +1,108 @@
+# Soft k-means clustering
+Soft k-means, also known as fuzzy clustering, is the probabilistic version of the better knwon k-means clustering.
+In this example, we simulate a one dimension clustering problem to illustrate a use-case of Chainsail.
+
+
+
+## Data generation
+We generated two 1-dimension clusters of points by drawing them from two normal probability distributions:
+- **Cluster 1:** 80 points, drawn for a normal probability distribution of mean `mu1 = 0` and standard deviation `sigma = 1`
+- **Cluster 2:** 120 points, drawn for a normal probability distribution of mean `mu2 = 1.5` and standard deviation `sigma = 1`
+
+<img alt="data points" src="./images/data_points_1D.png" height="400"/>
+
+
+
+## Model
+The model used to fit those clusters is a gaussian mixture (inspired from the [Stan example of soft k-means](https://mc-stan.org/docs/2_21/stan-users-guide/soft-k-means.html)).
+It is defined in the file `probability.py`.  
+
+- **Number of mixture components:** Just like in regular "hard" k-means, the users defines the number of clusters to find a priori. There are two clusters to find in this case, so the mixture model was defined with 2 components.
+- **Components weights:** Since the clusters do not contain the same number of points, weights were set on the components of the mixture at `w1 = 0.4` and `w2 = 0.6` (which fits the proportions of data points in each cluster).
+- **Parameters to fit:** To keep things simple, only the means of the components were set a variable parameters. The standard deviations were fixed at `sigma = 1` (which fits the data generation model), and components weights were also kept fixed.
+- **Prior parameters:** The prior parameters for both components of the gaussian mixture were arbitrarily set at mean `mu=0` and standard deviation `sigma = 5`.
+
+The log-posterior distribution is visualized below:
+
+<img alt="posterior" src="./images/posterior_1D.png" height="400"/>
+
+The posterior shows two modes, which correspond to the two possible positions of the mixture model to explain the generation of the clusters (Situation 1: components 1 and 2 of the gaussian mixture generated cluster 1 and 2 respectively. Situation 2 (reverse): components 1 and 2 generated clusters 2 and 1 respectively).
+But since the clusters contain a different number of points, and the mixture components have different weights, one situation is less likely to be true than the other.
```

Comment:
```suggestion
But since the clusters contain a different number of data points and the mixture components have different weights, one configuration is less likely to be correct than the other.
```